app/views/storage_locations/_storage_location_row.html.erb

```diff
@@ -3,15 +3,16 @@
   <td><%= storage_location.address %></td>
   <td><%= storage_location.square_footage %></td>
   <td><%= storage_location.warehouse_type %></td>
-  <td><%= @inventory ? @inventory.quantity_for(storage_location: storage_location.id) : storage_location.size %></td>
+  <td><%= @inventory_item_totals[storage_location.id] %></td>
   <td><%= number_to_currency(storage_location.inventory_total_value_in_dollars(@inventory)) %></td>
   <td class="text-right">
     <%= view_button_to storage_location %>
     <%= edit_button_to edit_storage_location_path(storage_location) %>
     <% if storage_location.discarded? %>
       <%= reactivate_button_to storage_location_reactivate_path(storage_location), { confirm: confirm_reactivate_msg(storage_location.name) } %>
     <% else %>
-      <%= deactivate_button_to storage_location_deactivate_path(storage_location), { confirm: confirm_deactivate_msg(storage_location.name) } %>
+      <%= deactivate_button_to storage_location_deactivate_path(storage_location),
+        { confirm: confirm_deactivate_msg(storage_location.name), enabled: @inventory_item_totals[storage_location.id].zero? } %>
```

Comment:
Calculating the inventory total is super quick since it's all in memory. I think I'd rather not have the extra instance variable and do this check live.

Reply:
Removed that new instance variable.